kythe/go/serving/pipeline/beamio/beamio.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package beamio provides Beam transformations for common IO patterns.
+package beamio
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"reflect"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime/exec"
+)
+
+func init() {
+	beam.RegisterType(reflect.TypeOf((*encodeKeyValue)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*KeyValue)(nil)).Elem())
+}
+
+// EncodeKeyValues encodes each PCollection of KVs into encoded KeyValues and
+// flattens all entries into a single PCollection.
+func EncodeKeyValues(s beam.Scope, tables ...beam.PCollection) beam.PCollection {
+	var encodings []beam.PCollection
+	for _, table := range tables {
+		encoded := beam.ParDo(s, &encodeKeyValue{beam.EncodedCoder{table.Coder()}}, table)
+		encodings = append(encodings, encoded)
+	}
+	return beam.Flatten(s, encodings...)
+}
+
+type encodeKeyValue struct{ Coder beam.EncodedCoder }
+
+func (e *encodeKeyValue) ProcessElement(key beam.T, val beam.U) (KeyValue, error) {
```

Comment:
Add doc comment

Reply:
I'm going to push back on this.  Exported methods should _generally_ be documented but when it comes to Beam, this would cause of lot of redundant comments.  This method **is required** to be exported by the Beam framework, but its receiver is not exported because it is an implementation detail (like an unexported function).  This ParDo is already commented through the exported `EncodeKeyValues` PTransform above.

Reply:
sgtm